x/likenft/keeper/nft_helper.go

```diff
@@ -0,0 +1,40 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/likecoin/likechain/x/likenft/types"
+)
+
+func (k Keeper) resolveValidClaimPeriod(ctx sdk.Context, classId string, classData types.ClassData, userAddress sdk.AccAddress) (*types.ClaimPeriod, error) {
+
+	claimPeriods := classData.Config.ClaimPeriods
+	if claimPeriods == nil {
+		return nil, types.ErrFailedToMintNFT.Wrapf(fmt.Sprintf("Pay to mint is not configurated for the class %s ", classId))
+	}
+
+	for _, claimPeriod := range claimPeriods {
+		// Check the first applicable claim period
+		if claimPeriod.StartTime.Before(ctx.BlockHeader().Time) {
+			// Check if the user is allowed to mint the token
+			// If allowed address list is nil it means the the class is publically available
+			if claimPeriod.AllowedAddressList == nil {
+				return claimPeriod, nil
+			}
+			for _, allowedAddress := range claimPeriod.AllowedAddressList {
+				// Ensure the configured allowed address is valid
+				wrappedAddress, err := sdk.AccAddressFromBech32(allowedAddress)
+				if err != nil {
+					return nil, types.ErrFailedToMintNFT.Wrapf(fmt.Sprintf("Failed to parse allowed address %s", allowedAddress))
```

Comment:
no need to throw? just continue loop?

Reply:
return should halt the entire function and stop the transaction, just to prevent potential mishandled transactions